### PR TITLE
Issue #2833: use editor arguments in InteractiveEditor

### PR DESCRIPTION
**Pull request by cabrinha**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to the README.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: util/cli/cli.go [hunk 1]
```diff
@@ -204,16 +204,21 @@ func setComments(input []byte, comments string) []byte {
...
+	var editorArgs []string
+	if strings.Contains(editor, " ") {
+		editorArgs = strings.Split(editor, " ")
+	}
 	errorComment := ""
 	for {
 		data = setComments(data, errorComment)
 		tempFile := writeToTempFile(filePattern, data)
-		cmd := exec.Command(editor, tempFile)
+		cmd := exec.Command(editor, editorArgs[1], tempFile)
```
Comment (alexmt) [inline util/cli/cli.go]: If no editor is specified then CLI might crash here due to nil pointer dereference. Also editor might has more than one argument. I would propose change it as following:

```
	var editor string
	var editorArgs []string
	if overrideEditor := os.Getenv(editorEnv); overrideEditor == "" {
		editor = defaultEditor
	} else {
		parts := strings.Fields(overrideEditor)
		editor = parts[0]
		editorArgs = parts[1:]
	}

	errorComment := ""
	for {
		data = setComments(data, errorComment)
		tempFile := writeToTempFile(filePattern, data)
		cmd := exec.Command(editor, append(editorArgs, tempFile)...)
```

@cabrinha let me know what you think please

**Review by alexmt — CHANGES_REQUESTED** [review]

File: util/cli/cli.go (hunk 1 above)
Comment (cabrinha) [reply util/cli/cli.go]: I really like this solution.

**Review by alexmt — APPROVED** [review]
LGTM!

**Comment by cabrinha** [comment]
Looks like these tests failed: 

```
--- FAIL: TestHookDeletePolicyHookSucceededHookExit0 (28.59s)
--- FAIL: TestHookDeletePolicyHookFailedHookExit0 (12.22s)
```
With the same message both times: 

```
msg=\"Timed out (10s) waiting for app \\\"test-hook-delete-policy-hook-succeeded-hook-exit0\\\"
```

**Comment by alexmt** [comment]
I suspect tests are flaky. Retrying to confirm

**Pull request merged**